### PR TITLE
Edit Department

### PR DESCRIPTION
**Pull request by powertashton**

### CODE REVIEW

File: Help Desk/helpDesk_editDepartment.php [hunk 1]
```diff
@@ -16,15 +16,44 @@
...
-if (!isActionAccessible($guid, $connection2, '/modules/Help Desk/helpDesk_editDepartment.php')) {
+if (!isActionAccessible($guid, $connection2, '/modules/Help Desk/helpDesk_manageDepartments.php')) {
     //Acess denied
     $page->addError(__('You do not have access to this action.'));
 } else {
     //Proceed!
+    $departmentID = $_GET['departmentID'] ?? '';
+
+    $departmentGateway = $container->get(DepartmentGateway::class);
+    $values = $departmentGateway->getByID($departmentID);
```
Comment (raynichc) [inline Help Desk/helpDesk_editDepartment.php:31]: You should add a check here for is values is empty. Have a look at edit classes for subcategories or techgroups for example

File: Help Desk/helpDesk_editDepartment.php [hunk 2]
```diff
@@ -16,15 +16,44 @@
...
+    
+    $form = Form::create('createDepartment',  $gibbon->session->get('absoluteURL') . '/modules/' . $gibbon->session->get('module') . '/helpDesk_editDepartmentProcess.php', 'post');
+    $form->addHiddenValue('address', $gibbon->session->get('address'));
+
+    $row = $form->addRow();
+        $row->addLabel('departmentName', __('Department Name'));
+        $row->addTextField('departmentName')
+            ->readOnly()
```
Comment (raynichc) [inline Help Desk/helpDesk_editDepartment.php:39]: Why is this read only?

Comment (raynichc) [reply Help Desk/helpDesk_editDepartment.php:39]: If its to keep the name unique, I can handle that.

Comment (powertashton) [reply Help Desk/helpDesk_editDepartment.php:39]: Yeah it's to keep the name unique, I'll let you handle it :)

File: Help Desk/helpDesk_editDepartment.php [hunk 3]
```diff
@@ -16,15 +16,44 @@
...
     //Acess denied
     $page->addError(__('You do not have access to this action.'));
 } else {
     //Proceed!
+    $departmentID = $_GET['departmentID'] ?? '';
+
+    $departmentGateway = $container->get(DepartmentGateway::class);
+    $values = $departmentGateway->getByID($departmentID);
+    
+    $form = Form::create('createDepartment',  $gibbon->session->get('absoluteURL') . '/modules/' . $gibbon->session->get('module') . '/helpDesk_editDepartmentProcess.php', 'post');
```
Comment (raynichc) [inline Help Desk/helpDesk_editDepartment.php:33]: Have you created the process page yet?

Comment (powertashton) [reply Help Desk/helpDesk_editDepartment.php]: ...I may have forgotten about it

File: Help Desk/helpDesk_editDepartment.php [hunk 4]
```diff
@@ -16,15 +16,49 @@
...
     //Acess denied
     $page->addError(__('You do not have access to this action.'));
 } else {
     //Proceed!
+    $departmentID = $_GET['departmentID'] ?? '';
+
+    $departmentGateway = $container->get(DepartmentGateway::class);
+    $values = $departmentGateway->getByID($departmentID);
+    
+    if (empty($groupID) || empty($values)) {
```
Comment (raynichc) [inline Help Desk/helpDesk_editDepartment.php:33]: This should be departmentID

**Review by raynichc — APPROVED** [review]

**Pull request merged**